### DISCUSSION TITLE
resize.c - AcquireResizeFilter : #pragma omp master request

### DISCUSSION THREAD

**Discussion by MootoolsSoftware** [ImageMagick/ImageMagick#6399] [Development]
### ImageMagick version

7.1.1-12

### Operating system, version and so on

Windows

### Description

Hello,

I am using imagemagick to create thumbnail from a c++ program.
In the program, images are updated in a thread through a :

`
#pragma omp parallel for num_threads(OMPGetMaxThreadCount()) schedule(dynamic, OMPGetScheduleChunkSize(size))
for (int i = 0; i < size; i++)
{
...
UpdateImage(); // Call ImageMagick to read the image
`

ImageMagick resize.c / AcquireResizeFilter use the problematic #pragma omp master which involves an OMP error.

![image](https://github.com/ImageMagick/ImageMagick/assets/15813085/c0f5800f-7d0f-45f7-ab0c-2eeac4ca8f90)

To avoid the problem I suggest to replace the master directive by a:

`
  /*
    Expert Option Request for verbose details of the resulting filter.
  */
  if (IsStringTrue(GetImageArtifact(image, "filter:verbose")) != MagickFalse)
  {
    // MANU CHANGES: move block
#if defined(MAGICKCORE_OPENMP_SUPPORT)
#pragma omp master
    {
#endif

`

At least, this make the code being nest in omp parallel for when filter:verbose is not used, which is the most common case.

Note that such problems can occurs on other portion of code: resample.c - SetResampleFilter an #pragma omp single

Thanks !
Manuel

**Comment by dlemstra** [comment]
I think we should change that to `#pragma omp single` instead @urban-warrior?

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT main branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://imagemagick.org/archive/beta/ by sometime tomorrow.
